File: Android/app/src/main/java/com/example/aquacontrol/SubtleDividerDecoration.java
package com.example.aquacontrol;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Paint;
import android.util.TypedValue;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.Objects;

public class SubtleDividerDecoration extends RecyclerView.ItemDecoration {
    private final Paint paint;
    private final int height;

    public SubtleDividerDecoration(Context context, int heightDp, int color) {
        this.height = (int) TypedValue.applyDimension(
                TypedValue.COMPLEX_UNIT_DIP, heightDp, context.getResources().getDisplayMetrics());

        paint = new Paint();
        paint.setColor(color);
    }

    @Override
    public void onDrawOver(@NonNull Canvas canvas, @NonNull RecyclerView parent, @NonNull RecyclerView.State state) {
        int childCount = parent.getChildCount();
        int itemCount = Objects.requireNonNull(parent.getAdapter()).getItemCount();

        for (int i = 0; i < childCount; i++) {
            View child = parent.getChildAt(i);
            int position = parent.getChildAdapterPosition(child);

            if (position == RecyclerView.NO_POSITION || position == itemCount - 1) {
                continue;
            }

            float top = child.getBottom();
            float bottom = top + height;

            canvas.drawRect(child.getLeft(), top, child.getRight(), bottom, paint);
        }
    }
}

File: Android/app/src/main/java/com/example/aquacontrol/SubtleDividerDecoration.kt
package com.example.aquacontrol

import android.content.Context
import android.graphics.Canvas
import android.graphics.Paint
import android.util.TypedValue
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.ItemDecoration
import java.util.Objects
import androidx.core.view.size

class SubtleDividerDecoration(context: Context, heightDp: Int, color: Int) : ItemDecoration() {
    private val paint: Paint
    private val height: Int

    init {
        this.height = TypedValue.applyDimension(
            TypedValue.COMPLEX_UNIT_DIP,
            heightDp.toFloat(),
            context.getResources().getDisplayMetrics()
        ).toInt()

        paint = Paint()
        paint.setColor(color)
    }

    override fun onDrawOver(canvas: Canvas, parent: RecyclerView, state: RecyclerView.State) {
        val adapter = parent.adapter ?: return
        val itemCount = adapter.itemCount
        val childCount = parent.childCount

        for (i in 0 until childCount) {
            val child = parent.getChildAt(i)
            val position = parent.getChildAdapterPosition(child)
            if (position == RecyclerView.NO_POSITION || position == itemCount - 1) {
                continue
            }

            val top = child.bottom.toFloat()
            val bottom = top + height

            canvas.drawRect(
                child.left.toFloat(),
                top,
                child.right.toFloat(),
                bottom,
                paint
            )
        }
    }
}
